substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -84,30 +65,23 @@ impl WasmInstance for Instance {
 			return (Err(format!("call into the runtime method '{name}': failed to write the input payload into guest memory: {error}").into()), None);
 		}
 
-		let mut state = ();
-		let mut call_args = polkavm::CallArgs::new(&mut state, method_index);
-		call_args.args_untyped(&[data_pointer, raw_data_length]);
-
-		match self.0.call(Default::default(), call_args) {
-			Ok(()) => {},
-			Err(polkavm::ExecutionError::Trap(trap)) => {
-				return (
-					Err(format!("call into the runtime method '{name}' failed: {trap}").into()),
-					None,
-				);
-			},
-			Err(polkavm::ExecutionError::Error(error)) => {
+		match self.0.call_typed(&mut (), name, (data_pointer, raw_data_length)) {
+			Ok(()) => (),
+			Err(polkavm::CallError::Trap) => (),
```

Comment:
This is incorrect; traps shouldn't be ignored. Restore the old arm body which returned an error.

Reply:
The previous version resulted compilation error. Thus the change.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6533/commits/0582e942d93817021de0c5228ffd6a65a90a11c2